miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -371,6 +374,49 @@ private void validateAliasUniqueness(Sample sample) throws IOException {
     }
   }
 
+  /**
+   * Adds any external names in the sample to the sample's identity
+   * 
+   * @param sample
+   * @throws IOException
+   */
+
+  private void addExternalNames(DetailedSample sample) throws IOException {
+    if (sample.getExternalNames() == null) return;
+
+    DetailedSample child = sample;
+    DetailedSample parent = sample.getParent();
+    while (parent.getParent() != null) {
+      child = parent;
+      parent = child.getParent();
+    }
+
+    if (!isIdentitySample(parent)) {
+      throw new IllegalStateException("Missing Identity at root of hierarchy");
+    }
+
+    SampleIdentity identity = (SampleIdentity) parent;
+    List<String> identityExternalNames = Arrays.asList(identity.getExternalName().split(",")).stream().map(String::trim)
+        .collect(Collectors.toList());
+    List<String> sampleExternalNames = Arrays.asList(sample.getExternalNames().split(",")).stream().map(String::trim)
+        .collect(Collectors.toList());
+    Set<String> lowerCaseIdentityExternalNames = identityExternalNames.stream().map(String::toLowerCase).collect(Collectors.toSet());
+    sampleExternalNames.stream().forEach(name -> {
+      try {
+        if (!lowerCaseIdentityExternalNames.contains(name.toLowerCase()) && !(isUniqueExternalNameWithinProjectRequired() &&
+            getIdentitiesByExternalNameOrAliasAndProject(name, identity.getProject().getId(), true).size() > 0)) {
+          identityExternalNames.add(name);
+        }
+      } catch (IOException e) {
+        log.error("Failed to retrieve all external names: " + e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)